bentoml/yatai/client/bento_repository_api.py

```diff
@@ -342,65 +344,113 @@ def list(
+    ):
         """
-        Delete all BentoServices that matches the specified criteria
+        Delete bentos that matches the specified criteria
 
         Args:
-            bento_name: optional
-            labels: optional
-
+            bento_tag: string
+            labels: string
+            bento_name: string
+            bento_version: string
+            prune: boolean, Set True to delete all BentoService
+            require_confirm: boolean
         Example:
-
+        >>>
         >>> yatai_client = get_yatai_client()
         >>> # Delete all bento services
-        >>> yatai_client.repository.prune()
-        >>> # Delete bento services that matches with the label `ci=failed`
-        >>> yatai_client.repository.prune(labels='ci=failed')
+        >>> yatai_client.repository.delete(prune=True)
+        >>> # Delete bento service with name is `IrisClassifier` and version `0.1.0`
+        >>> yatai_client.repository.delete(
+        >>>     bento_name='IrisClassifier', bento_version='0.1.0'
+        >>> )
+        >>> # or use bento tag
+        >>> yatai_client.repository.delete('IrisClassifier:v0.1.0')
+        >>> # Delete all bento services with name 'MyService`
+        >>> yatai_client.repository.delete(bento_name='MyService')
+        >>> # Delete all bento services with labels match `ci=failed` and `cohort=20`
+        >>> yatai_client.repository.delete(labels='ci=failed, cohort=20')
         """
-        track('py-api-prune')
-        list_bentos_result = self.list(bento_name=bento_name, labels=labels,)
-        if list_bentos_result.status.status_code != yatai_proto.status_pb2.Status.OK:
-            error_code, error_message = status_pb_to_error_code_and_message(
-                list_bentos_result.status
+        track('py-api-delete')
+
+        delete_list_limit = 50
+
+        if (
+            bento_tag is not None
+            and bento_name is not None
+            and bento_version is not None
+        ):
+            raise BentoMLException('Too much arguments')
+
+        if bento_tag is not None:
+            logger.info(f'Deleting saved Bento bundle {bento_tag}')
+            return self._delete_bento_bundle(bento_tag, require_confirm)
+        elif bento_name is not None and bento_tag is not None:
+            logger.info(f'Deleting saved Bento bundle {bento_name}:{bento_version}')
+            return self._delete_bento_bundle(
+                f'{bento_name}:{bento_version}', require_confirm
             )
-            raise BentoMLException(f'{error_code}:{error_message}')
-        for bento in list_bentos_result.bentos:
-            bento_tag = f'{bento.name}:{bento.version}'
-            try:
-                self.delete(bento_tag)
-            except BentoMLException as e:
-                logger.error(f'Failed to delete Bento {bento_tag}: {e}')
+        else:
+            # list of bentos
+            if prune is True:
+                logger.info('Deleting all BentoML saved bundles.')
```

Comment:
if local:
   log: "Deleting all bento bundles in local" 
elif remote:
   log:  "Deleting all bento bundles in YataiService at {yatai_url}"